azure-samples/src/main/java/com/microsoft/azure/management/network/samples/ManageNetworkWatcher.java

```diff
@@ -0,0 +1,340 @@
+            // Verify IP flow – verify if traffic is allowed to or from a virtual machine
+            // Get the IP address of a NIC on a virtual machine
+            String ipAddress = vm.getPrimaryNetworkInterface().ipConfigurations().values().iterator().next().privateIPAddress();
+            // Test IP flow on the NIC
+            System.out.println("Verifying IP flow for vm id " + vm.id() + "...");
+            VerificationIPFlow verificationIPFlow = nw.verifyIPFlow()
+                    .withTargetResourceId(vm.id())
+                    .withDirection(Direction.INBOUND)
+                    .withProtocol(Protocol.TCP)
+                    .withLocalIPAddress(ipAddress)
+                    .withRemoteIPAddress("8.8.8.8")
+                    .withLocalPort("443")
+                    .withRemotePort("443")
+                    .execute();
+            Utils.print(verificationIPFlow);
+
+            //============================================================
+            // Analyze next hop – get the next hop type and IP address for a virtual machine
+            System.out.println("Calculating next hop...");
+            NextHop nextHop = nw.nextHop().withTargetResourceId(vm.id())
+                    .withSourceIPAddress(ipAddress)
+                    .withDestinationIPAddress("8.8.8.8")
+                    .execute();
+            Utils.print(nextHop);
+
+            //============================================================
+            // Retrieve network topology for a resource group
+            System.out.println("Getting topology...");
+            Topology topology = nw.getTopology(rgName);
+            Utils.print(topology);
+
+            //============================================================
+            // Analyze Virtual Machine Security by examining effective network security rules applied to a VM
+            // Get security group view for the VM
+            System.out.println("Getting security group view for a vm");
+            SecurityGroupView sgViewResult = nw.getSecurityGroupView(vm.id());
+            Utils.print(sgViewResult);
+
+            //============================================================
+            // Configure Network Security Group Flow Logs
+
+            // Get flow log settings
+            FlowLogSettings flowLogSettings = nw.getFlowLogSettings(nsg.id());
+            Utils.print(flowLogSettings);
+
+            // Enable NSG flow log
+            flowLogSettings.update()
+                    .withLogging()
+                    .withStorageAccount(storageAccount.id())
+                    .withRetentionPolicyDays(5)
+                    .withRetentionPolicyEnabled()
+                    .apply();
+            Utils.print(flowLogSettings);
+
+            // wait for flow log to log an event
+            System.out.println("Waiting for flow log to log an event...");
+            Thread.sleep(250000);
```

Comment:
let's not hard-code the delay into the sample this way, because then it will slow down the playback in CI as well -- use `SdkContext.sleep(...)` instead.